orchagent/fdborch.cpp

```diff
@@ -0,0 +1,76 @@
+#include <assert.h>
+#include <iostream>
+
+#include "logger.h"
+#include "fdborch.h"
+
+extern sai_fdb_api_t *sai_fdb_api;
+
+void FdbOrch::update(sai_fdb_event_t type, const sai_fdb_entry_t* entry, sai_object_id_t portOid)
+{
+    SWSS_LOG_ENTER();
+
+    FdbUpdate update;
+    update.entry.mac = entry->mac_address;
+    update.entry.vlan = entry->vlan_id;
+
+    switch (type)
+    {
+    case SAI_FDB_EVENT_LEARNED:
+        if (!m_portsOrch->getPort(portOid, update.port))
+        {
+            SWSS_LOG_ERROR("Failed to get port for %lu OID\n", portOid);
+            return;
+        }
+
+        update.add = true;
+        break;
+    case SAI_FDB_EVENT_AGED:
+    case SAI_FDB_EVENT_FLUSHED:
+        update.add = false;
+        break;
+    }
+
+    for (auto observer: m_observers)
+    {
+        observer->update(SUBJECT_TYPE_FDB_CHANGE, static_cast<void *>(&update));
+    }
+}
+
+bool FdbOrch::getPort(const MacAddress& mac, uint16_t vlan, Port& port)
+{
+    SWSS_LOG_ENTER();
+
+    sai_fdb_entry_t entry;
+    memcpy(entry.mac_address, mac.getMac(), sizeof(sai_mac_t));
+    entry.vlan_id = vlan;
+
+    sai_attribute_t *attr = NULL;
+    sai_attribute_t attrs[SAI_FDB_ENTRY_ATTR_END];
+
+    sai_status_t status = sai_fdb_api->get_fdb_entry_attribute(&entry, SAI_FDB_ENTRY_ATTR_END, attrs);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_INFO("Failed to get port for FDB entry OID\n");
+        return false;
+    }
```

Comment:
since we are only want to get the port attribute, I think we can do

attr.id = SAI_FDB_ENTRY_ATTR_PORT_ID

get_fdb_entry_attribute(&etnry, 1, attr).

by the way, user are required to give attr.id in order to get the attribute value via sai api.